include/aspect/geometry_model/chunk.h

```diff
@@ -0,0 +1,205 @@
+/*
+  Copyright (C) 2011, 2012 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__geometry_model_chunk_h
+#define __aspect__geometry_model_chunk_h
+
+#include <aspect/geometry_model/interface.h>
+#include <deal.II/grid/manifold.h>
+#include <deal.II/base/function_lib.h>
+#include <deal.II/grid/grid_out.h>
+
+namespace
+{
+  using namespace dealii;
+
+  /**
+   * A geometry model class that describes a chunk of a spherical shell.
+   * In 2D, this class simulates a sector with inner and outer radius, and
+   * minimum and maximum longitude. Longitude increases anticlockwise
+   * from the +ve x-axis, as per the mathematical convention of phi.
+   * In 3D, the class simulates a chunk of a sphere, bounded by arbitrary
+   * lines of longitude, latitude and radius. Boundary indicator names are
+   * west, east, south, north, inner and outer.
+   *
+   * The parameters that describe this geometry and that are read from the
+   * input file are the inner and outer radii of the shell, the minimum
+   * and maximum longitude, minimum and maximum longitude, and the
+   * number of cells initialised in each dimension.
+   */
+
+  template <int dim>
+  class ChunkGeometry : public ChartManifold<dim,dim>
+  {
+    public:
+      virtual
+      Point<dim>
+      pull_back(const Point<dim> &space_point) const;
```

Comment:
Ok, I saw that they are actually used and important, please add a short comment anyway